cmd/soci/commands/internal/snapshotter.go

```diff
@@ -43,3 +47,25 @@ var SnapshotterFlags = []cli.Flag{
 		EnvVar: "CONTAINERD_SNAPSHOTTER",
 	},
 }
+
+func SnapshotterRootPathFromContext(c *cli.Context) string {
+	if root := c.GlobalString("root"); root != "" {
+		return root
+	}
+	return config.DefaultSociSnapshotterRootPath
+}
+
+func EnsureSnapshotterRootPath(c *cli.Context) error {
+	root := SnapshotterRootPathFromContext(c)
+	// Creating the snapshotter's root path first if it does not exist, since this ensures, that
+	// it has the limited permission set as drwx--x--x.
+	// The subsequent oci.New creates a root path dir with too broad permission set.
+	if _, err := os.Stat(root); os.IsNotExist(err) {
```

Comment:
Can we move this part into the soci package somewhere and call it from the snapshotter as well?

The snapshotter implicitly creates this directory when it creates the socket, but it uses 0700 which I think is more appropriate.